### ISSUE TITLE
Modifying on a copy

### ISSUE THREAD

**Issue by Saqibm128** [YerevaNN/mimic3-benchmarks#24]
Running preprocessing.clean_events, specifically this line

`events.VALUE.ix[idx] = clean_fn(events.ix[idx])`

gives back an error suggesting that we may be modifying a copy of events instead of events itself. The function returns events, which is the original parameter passed in. However, if we modify a copy instead, then the original events object is essentially untouched during this process.

To summarize, this may be problematic, as it may suggest that the clean functions do not actually return their results correctly.
http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy

**Comment by turambar** [comment]
Yeah, good catch. There are a few places where we use operations that generate these warnings and could turn into real problems for future versions of pandas. If you can help us identify (and even fix) these cases, that would be a big help!

**Comment by turambar** [comment]
Addressed in https://github.com/YerevaNN/mimic3-benchmarks/pull/32

**Comment by turambar** [comment]
Merged https://github.com/YerevaNN/mimic3-benchmarks/pull/32
